### PR TITLE
first improvements to BayesianDenseLayer code & docstrings

### PR DESCRIPTION
**Pull request by st--**
See #19

### CODE REVIEW

File: gpflux/layers/bayesian_dense_layer.py [hunk 1]
```diff
@@ -128,25 +139,14 @@ def predict_samples(
         inputs: TensorType,
         *,
         num_samples: Optional[int] = None,
-        full_output_cov: bool = False,
-        full_cov: bool = False,
-        whiten: bool = False,
```
Comment (st--) [inline gpflux/layers/bayesian_dense_layer.py:133]: not used anywhere

File: gpflux/layers/bayesian_dense_layer.py [hunk 2]
```diff
@@ -89,7 +101,6 @@ def __init__(
         self.activation = activation
         self.is_mean_field = is_mean_field
         self.temperature = temperature
-        self.returns_samples = returns_samples
```
Comment (st--) [inline gpflux/layers/bayesian_dense_layer.py:92]: not used anywhere (except in one part of a test, which I removed)

File: gpflux/layers/bayesian_dense_layer.py [hunk 3]
```diff
@@ -43,30 +53,32 @@ def __init__(
         w_sqrt: Optional[np.ndarray] = None,
         activation: Optional[Callable] = None,
         is_mean_field: bool = True,
-        temperature: float = 1e-4,
-        returns_samples: bool = True,
+        temperature: float = 1e-4,  # TODO is this intentional?
```
Comment (st--) [inline gpflux/layers/bayesian_dense_layer.py]: just to flag up that it seems weird to initialize the temperature to such a small value

File: gpflux/layers/bayesian_dense_layer.py [hunk 4]
```diff
@@ -43,30 +53,32 @@ def __init__(
...
+            If not specified, this defaults to `xavier_initialization_numpy`
+            for the weights and zero for the bias.
+        :param w_sqrt: Initial value of the variational Cholesky of the
+            (co)variance for weights + bias. If not specified, this defaults to
+            1e-5 * Identity.
+        :param activation: The activation function. If not specified, this defaults to the identity.
+        :param is_mean_field: Determines whether the approximation to the
+            weight posterior is mean field. Must be consistent with the shape
+            of ``w_sqrt``, if specified.
+        :param temperature: For cooling (< 1.0) or heating (> 1.0) the posterior.
```
Comment (st--) [inline gpflux/layers/bayesian_dense_layer.py]:
```suggestion
        :param temperature: The KL loss will be scaled by this factor.
            Can be used for cooling (< 1.0) or heating (> 1.0) the posterior.
```

**Review by vdutor — APPROVED** [review]

File: gpflux/layers/bayesian_dense_layer.py (hunk 3 above)
Comment (st--) [reply gpflux/layers/bayesian_dense_layer.py]: @vdutor should this stay at 0.0001 or change it to 1.0?

Comment (st--) [reply gpflux/layers/bayesian_dense_layer.py]: @fleibfried what was your thinking behind the default value ?

File: gpflux/layers/bayesian_dense_layer.py (hunk 4 above)
Comment (st--) [reply gpflux/layers/bayesian_dense_layer.py]: Should also mention what the default value is in here

File: gpflux/layers/bayesian_dense_layer.py (hunk 3 above)
Comment (fleibfried) [reply gpflux/layers/bayesian_dense_layer.py]: I think back in the days I have read somewhere that BNNs practically perform better when cooling the posterior by downweighting the KL. A default setting of 1 is not expected to yield practically best results I believe.

Comment (st--) [reply gpflux/layers/bayesian_dense_layer.py]: that makes sense - is there some citation we could add why it should be 1e-4? or how did you choose this value in particular?

Comment (fleibfried) [reply gpflux/layers/bayesian_dense_layer.py]: it was after reading "How Good is the Bayes Posterior in Deep Neural Networks Really?" from Wenzel et al. 2020:

https://arxiv.org/abs/2002.02405

They have multiple examples where they plot evaluation metrics vs. temperature. I have just skimmed through them and maybe something in between 10^-3 and 10^-2 would be a good default initialization based on their empirical results (10^-4 seems not too bad though).

**Pull request merged**